infrastructure/smart-contracts/contracts/upgrades/referee/Referee10.sol

```diff
@@ -0,0 +1,1117 @@
+        bytes activeChallengerPublicKey; // The challengerPublicKey that was active at the time of challenge submission
+        address rollupUsed; // The rollup address used for this challenge
+        uint256 createdTimestamp; // used to determine if a node license is eligible to submit
+        uint256 totalSupplyOfNodesAtChallengeStart; // keep track of what the total supply opf nodes is when the challenge starts
+        uint256 rewardAmountForClaimers; // this is how much esXai should be allocated to the claimers
+        uint256 amountForGasSubsidy; // this is how much Xai was minted for the gas subsidy
+        uint256 numberOfEligibleClaimers; // how many submitters are eligible for claiming, used to determine the reward amount
+        uint256 amountClaimedByClaimers; // keep track of how much Xai has been claimed by the claimers, primarily used to expire unclaimed rewards 
+    }
+
+    // Define events
+    event ChallengeSubmitted(uint256 indexed challengeNumber);
+    event ChallengeClosed(uint256 indexed challengeNumber);
+    event AssertionSubmitted(uint256 indexed challengeId, uint256 indexed nodeLicenseId);
+    event RollupAddressChanged(address newRollupAddress);
+    event ChallengerPublicKeyChanged(bytes newChallengerPublicKey);
+    event NodeLicenseAddressChanged(address newNodeLicenseAddress);
+    event AssertionCheckingToggled(bool newState);
+    event Approval(address indexed owner, address indexed operator, bool approved);
+    event KycStatusChanged(address indexed wallet, bool isKycApproved);
+    event InvalidSubmission(uint256 indexed challengeId, uint256 nodeLicenseId);
+    event InvalidBatchSubmission(uint256 indexed challengeId, address submissionAddress, address operator, uint256 keysLength);
+    event RewardsClaimed(uint256 indexed challengeId, uint256 amount);
+    event BatchRewardsClaimed(uint256 indexed challengeId, uint256 totalReward, uint256 keysLength);
+    event BulkRewardsClaimed(uint256 indexed challengeId, address indexed bulkAddress, uint256 totalReward, uint256 winningKeys);
+    event ChallengeExpired(uint256 indexed challengeId);
+    event StakingEnabled(bool enabled);
+    event UpdateMaxStakeAmount(uint256 prevAmount, uint256 newAmount);
+    event UpdateMaxKeysPerPool(uint256 prevAmount, uint256 newAmount);
+    event StakedV1(address indexed user, uint256 amount, uint256 totalStaked);
+    event UnstakeV1(address indexed user, uint256 amount, uint256 totalStaked);
+    event BatchChallenge(uint256 indexed challengeId, uint64[] assertionIds);
+    event NewBulkSubmission(uint256 indexed challengeId, address indexed bulkAddress, uint256 stakedKeys, uint256 winningKeys);
+    event UpdateBulkSubmission(uint256 indexed challengeId, address indexed bulkAddress, uint256 stakedKeys, uint256 winningKeys, uint256 increase, uint256 decrease);
+
+    function initialize() public reinitializer(8) {
+
+        maxStakeAmountPerLicense = 200 * 10 ** 18;
+        maxKeysPerPool = 100_000;
+
```

Comment:
This is missing the update of the tier thresholds.

We do not have to update the boostFactors as they will not change to how Referee8 was initialized.

```
        stakeAmountTierThresholds[0] = 30_000 * 10 ** 18;
        stakeAmountTierThresholds[1] = 2_000_000 * 10 ** 18;
        stakeAmountTierThresholds[2] = 4_000_000 * 10 ** 18;
        stakeAmountTierThresholds[3] = 8_000_000 * 10 ** 18;
```